NetConsole/netconsole.cpp

```diff
@@ -0,0 +1,420 @@
+// Output : true on success, false otherwise
+//-----------------------------------------------------------------------------
+bool CNetCon::Init(void)
+{
+	WSAData wsaData{};
+	int nError = ::WSAStartup(MAKEWORD(2, 2), &wsaData);
+
+	if (nError != 0)
+	{
+		spdlog::error("Failed to start Winsock via WSAStartup: ({})", NET_ErrorString(WSAGetLastError()));
+		return false;
+	}
+
+	this->TermSetup();
+
+	std::thread tFrame(&CNetCon::RunFrame, this);
+	tFrame.detach();
+
+	return true;
+}
+
+//-----------------------------------------------------------------------------
+// Purpose: WSA and NETCON systems shutdown
+// Output : true on success, false otherwise
+//-----------------------------------------------------------------------------
+bool CNetCon::Shutdown(void)
+{
+	m_pSocket->CloseAllAcceptedSockets();
+	m_abConnEstablished = false;
+
+	int nError = ::WSACleanup();
+	if (nError != 0)
+	{
+		spdlog::error("Failed to stop winsock via WSACleanup: ({})", NET_ErrorString(WSAGetLastError()));
+		return false;
+	}
+	return true;
+}
+
+//-----------------------------------------------------------------------------
+// Purpose: terminal setup
+//-----------------------------------------------------------------------------
+void CNetCon::TermSetup(void)
+{
+	DWORD dwMode = NULL;
+	HANDLE hInput = GetStdHandle(STD_INPUT_HANDLE);
+	HANDLE hOutput = GetStdHandle(STD_OUTPUT_HANDLE);
+
+	CONSOLE_SCREEN_BUFFER_INFOEX sbInfoEx{};
+	COLORREF storedBG = sbInfoEx.ColorTable[0];
+	sbInfoEx.cbSize = sizeof(CONSOLE_SCREEN_BUFFER_INFOEX);
+
+	GetConsoleScreenBufferInfoEx(hOutput, &sbInfoEx);
+	sbInfoEx.ColorTable[0] = 0x0000;
+	SetConsoleScreenBufferInfoEx(hOutput, &sbInfoEx);
+}
+
+//-----------------------------------------------------------------------------
+// Purpose: gets input IP and port for initialization
+//-----------------------------------------------------------------------------
+void CNetCon::UserInput(void)
+{
+	std::string svInput;
+
+	if (std::getline(std::cin, svInput))
+	{
+		if (strcmp(svInput.c_str(), "nquit") == 0)
+		{
+			m_bQuitApplication = true;
+			return;
+		}
+		if (m_abConnEstablished)
+		{
+			if (strcmp(svInput.c_str(), "disconnect") == 0)
+			{
+				this->Disconnect();
+				return;
+			}
+			size_t nPos = svInput.find(" ");
+			if (!svInput.empty() && nPos > 0 && nPos < svInput.size() && nPos != svInput.size())
+			{
+				std::string svSecondArg = svInput.substr(nPos + 1);
+				std::string svFirstArg = svInput;
+				svFirstArg = svFirstArg.erase(svFirstArg.find(" "));
+
+				if (strcmp(svFirstArg.c_str(), "PASS") == 0) // Auth with RCON server.
+				{
+					std::string svSerialized = this->Serialize(svSecondArg, "", cl_rcon::request_t::SERVERDATA_REQUEST_AUTH);
+					this->Send(svSerialized);
+				}
+				else if (strcmp(svFirstArg.c_str(), "SET") == 0) // Set value query.
+				{
+					std::string svSerialized = this->Serialize(svFirstArg, svSecondArg, cl_rcon::request_t::SERVERDATA_REQUEST_SETVALUE);
```

Comment:
I think there's a bug here, `requestBuf` will always be "SET" so the server will try to set the `SET` convar. Probably need to split the second arg into the var name and value?

Reply:
Thanks, probably a small mistake. I will take a look at this tomorrow.

Reply:
This bug has been fixed as of commit 566d3dc33b9c8be39efe429f8ca848f3ff7580ea.